wiring/inc/spark_wiring_watchdog.h

```diff
@@ -108,5 +106,195 @@ inline void application_checkin() { ApplicationWatchdog::checkin(); }
 
 inline void application_checkin() {  }
 
-#endif
+#endif // PLATFORM_THREADING
+
+
+#if Wiring_Watchdog
+
+#include "watchdog_hal.h"
+#include "enumflags.h"
+
+namespace particle {
+
+enum class WatchdogCaps : uint32_t {
+    NONE            = WATCHDOG_CAPS_NONE,
+    SOFT_RESET      = WATCHDOG_CAPS_SOFT_RESET,         /** The watchdog can be configured to not hard resetting device on expired. */
+    INT             = WATCHDOG_CAPS_INT,                /** The watchdog can generate an interrupt on expired. */
+    RECONFIGURABLE  = WATCHDOG_CAPS_RECONFIGURABLE,     /** The watchdog can be re-configured after started. */
+    STOPPABLE       = WATCHDOG_CAPS_STOPPABLE,          /** The watchdog can be stopped after started. */
+    ALL             = WATCHDOG_CAPS_ALL
+};
+ENABLE_ENUM_CLASS_BITWISE(WatchdogCaps);
+
+using WatchdogOnExpiredStdFunction = std::function<void(void)>;
+typedef void (*WatchdogOnExpiredCallback)(void* context);
+
+class WatchdogInfo : public hal_watchdog_info_t {
+public:
+    WatchdogInfo() {
+        this->size = sizeof(hal_watchdog_info_t);
+        this->version = HAL_WATCHDOG_VERSION;
+    }
+};
+
+class WatchdogConfiguration {
+public:
+    WatchdogConfiguration()
+            : config_() {
+        config_.size = sizeof(hal_watchdog_config_t);
+        config_.version = HAL_WATCHDOG_VERSION;
+        config_.timeout_ms = WATCHDOG_DEFAULT_TIMEOUT_MS;
```

Comment:
I'd avoid setting a default timeout, let the caller be explicit about it, 5s also doesn't sound like enough for most of the cases.